trading_calendars/calendar_helpers.py

```diff
@@ -44,28 +64,43 @@ def is_open(opens, closes, minute_val):
             return False
 
 
-def compute_all_minutes(opens_in_ns, closes_in_ns):
+def compute_all_minutes(
+    opens_in_ns, break_starts_in_ns, break_ends_in_ns, closes_in_ns,
+):
     """
-    Given arrays of opens and closes, both in nanoseconds,
-    return an array of each minute between the opens and closes.
-    """
-    deltas = closes_in_ns - opens_in_ns
-
-    # + 1 because we want 390 mins per standard day, not 389
-    daily_sizes = (deltas // NANOSECONDS_PER_MINUTE) + 1
-    num_minutes = daily_sizes.sum()
+    Given arrays of opens and closes (in nanoseconds) and optionally
+    break_starts and break ends, return an array of each minute between the
+    opens and closes.
 
-    # One allocation for the entire thing. This assumes that each day
-    # represents a contiguous block of minutes.
-    pieces = []
-
-    for open_, size in zip(opens_in_ns, daily_sizes):
-        pieces.append(
-            np.arange(open_,
-                      open_ + size * NANOSECONDS_PER_MINUTE,
-                      NANOSECONDS_PER_MINUTE)
-        )
-
-    out = np.concatenate(pieces).view('datetime64[ns]')
-    assert len(out) == num_minutes
+    NOTE: An extra minute is added to ending pbountaries (break_end and close)
+    so we include the last bar.
+    """
+    pieces = []  # todo preallocat?
+    for open_time, break_start_time, break_end_time, close_time in zip(
+        opens_in_ns, break_starts_in_ns, break_ends_in_ns, closes_in_ns
+    ):
+        if break_start_time != NP_NAT and break_end_time != NP_NAT:
```

Comment:
I don't think this will have performance implications, but I'll add a check at calendar construction that will raise an error if you have unmatched break starts/ends. It will be clearer.